Fix for the following error, apparently due to jammy's older version of
bindgen (0.59.1):

  FAILED: obj/build/rust/allocator/libbuild_srust_sallocator_callocator.rlib
  error[E0308]: arguments to this function are incorrect
    --> ../../build/rust/allocator/lib.rs:40:22
     |
  40 |             unsafe { ffi::alloc(layout.size(), layout.align()) }
     |                      ^^^^^^^^^^ -------------  -------------- expected `u64`, found `usize`
     |                                 |
     |                                 expected `u64`, found `usize`
     |
  note: function defined here
    --> ./../../build/rust/../../out/Release/gen/build/rust/allocator/allocator_impls_ffi/bindings.rs:3:609

  error[E0308]: arguments to this function are incorrect
    --> ../../build/rust/allocator/lib.rs:45:17

  ...

--- a/build/rust/allocator/lib.rs
+++ b/build/rust/allocator/lib.rs
@@ -35,23 +35,25 @@ mod cpp_allocator {
 
     struct Allocator;
 
+    pub type size_t = std::os::raw::c_ulong;  // XtraDeb fix
+
     unsafe impl GlobalAlloc for Allocator {
         unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
-            unsafe { ffi::alloc(layout.size(), layout.align()) }
+            unsafe { ffi::alloc(layout.size() as size_t, layout.align() as size_t) }
         }
 
         unsafe fn dealloc(&self, ptr: *mut u8, layout: Layout) {
             unsafe {
-                ffi::dealloc(ptr, layout.size(), layout.align());
+                ffi::dealloc(ptr, layout.size() as size_t, layout.align() as size_t);
             }
         }
 
         unsafe fn alloc_zeroed(&self, layout: Layout) -> *mut u8 {
-            unsafe { ffi::alloc_zeroed(layout.size(), layout.align()) }
+            unsafe { ffi::alloc_zeroed(layout.size() as size_t, layout.align() as size_t) }
         }
 
         unsafe fn realloc(&self, ptr: *mut u8, layout: Layout, new_size: usize) -> *mut u8 {
-            unsafe { ffi::realloc(ptr, layout.size(), layout.align(), new_size) }
+            unsafe { ffi::realloc(ptr, layout.size() as size_t, layout.align() as size_t, new_size as size_t) }
         }
     }
 
